### PR TITLE
feat(St-14): Add all entities according to ERD 

### PR DESCRIPTION
**Pull request by tareq-mahmood-sazim**
## Description of Change

- Removed previous changes as ERD has changed
- Created entities following new ERD
- Slight changes over controllers to match ERD
- Removed dependabot

## Please list which modules are affected by your changes

common/entities/**.ts, users/**.ts

---

- [x] **You assure that you have tested the modules properly and there are no breaking changes**
- [x] **You adhere to Sazim's [Engineering Principles](https://docs.google.com/document/d/10mowIwsuYMyNkXS7aeL7AJyVNMYVh1vdMf7yn-A6I1k/edit#heading=h.qqetrc8w2lm8)**

## Associated Ticket Link(s)

- https://trello.com/c/g2qYTfYr

## Related Pull Request(s)

- None

### CODE REVIEW

**Comment by nur-zaman-sazim** [comment]
There are duplicate changes in this PR and #18  you might need to change the pointer to the branch

File: src/users/users.dtos.ts [hunk 1]
```diff
@@ -25,6 +27,18 @@ export class UserProfileDto implements Pick<UserProfile, "firstName" | "lastName
...
+  educationLevel?: EducationLevel;
+
+  @IsString()
+  @MinLength(2)
+  @MaxLength(255)
+  majorSubject?: string;
+
+  @IsOptional()
+  @IsInt()
+  role?: number;
```
Comment (nur-zaman-sazim) [inline src/users/users.dtos.ts:41]: ooc: why are we using role as a number ?

File: scripts/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import postgres from "postgres";
+import * as readline from "readline";
+import { v4 } from "uuid";
+
+const sql = postgres("postgresql://postgres:postgres@localhost:5432/project_dev_db");
```
Comment (nur-zaman-sazim) [inline scripts/src/index.ts:5]: can we get this from .env ?

**Review by nur-zaman-sazim — CHANGES_REQUESTED** [review]

File: scripts/src/index.ts (hunk 2 above)
Comment (tareq-mahmood-sazim) [reply scripts/src/index.ts:5]: Yes bhaia this has been fixed

File: src/users/users.dtos.ts (hunk 1 above)
Comment (tareq-mahmood-sazim) [reply src/users/users.dtos.ts:41]: To handle permission levels, the higher the number is, the lower the permissions will be.

**Review by nur-zaman-sazim — APPROVED** [review]

**Pull request merged**